### ISSUE TITLE
Issues compiling FastNoise library on Linux

### ISSUE THREAD

**Issue by The64thGamer** [zak-grumbles/gd_fastnoise2#1]
There are issues generating the custom FastNoise2 library on Linux systems. Instead of compiling "FastNoise.lib", Linux will generate "libFastNoise.a".  This will cause the compiler to miss the file.

```
[ 96%] scons: *** [bin/godot.linuxbsd.template_release.x86_64.mono] Implicit dependency `modules/gd_fastnoise2/FastNoise2/src/FastNoise.lib' not found, needed by target `bin/godot.linuxbsd.template_release.x86_64.mono'.
scons: building terminated because of errors.
[Time elapsed: 00:00:08.676]
```

In addition, renaming the file to FastNoise.lib will not yield any results on compile as it will not find any functions in the library.

```
[ 96%] Linking Program bin/godot.linuxbsd.template_release.x86_64.mono ...
[ 99%] progress_finish(["progress_finish"], [])
[100%] /usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_generator.linuxbsd.template_release.x86_64.o): in function `FNGenerator::_init_node()':
fn_generator.cpp:(.text+0x42a): undefined reference to `FastNoise::Value* FastSIMD::New<FastNoise::Value>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: fn_generator.cpp:(.text+0x49a): undefined reference to `FastNoise::Perlin* FastSIMD::New<FastNoise::Perlin>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: fn_generator.cpp:(.text+0x50a): undefined reference to `FastNoise::CellularValue* FastSIMD::New<FastNoise::CellularValue>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: fn_generator.cpp:(.text+0x57a): undefined reference to `FastNoise::CellularDistance* FastSIMD::New<FastNoise::CellularDistance>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: fn_generator.cpp:(.text+0x5e2): undefined reference to `FastNoise::CellularLookup* FastSIMD::New<FastNoise::CellularLookup>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: fn_generator.cpp:(.text+0x654): undefined reference to `FastNoise::Simplex* FastSIMD::New<FastNoise::Simplex>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_generator.linuxbsd.template_release.x86_64.o): in function `FNGenerator::gen_uniform_2D_image(int, int, int, int, float, int) const':
fn_generator.cpp:(.text+0x158c): undefined reference to `FastNoise::ConvertRGBA8* FastSIMD::New<FastNoise::ConvertRGBA8>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_basic_generators.linuxbsd.template_release.x86_64.o): in function `FNConstant::FNConstant()':
fn_basic_generators.cpp:(.text+0x67): undefined reference to `FastNoise::Constant* FastSIMD::New<FastNoise::Constant>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_basic_generators.linuxbsd.template_release.x86_64.o): in function `FNWhite::FNWhite()':
fn_basic_generators.cpp:(.text+0x127): undefined reference to `FastNoise::White* FastSIMD::New<FastNoise::White>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_basic_generators.linuxbsd.template_release.x86_64.o): in function `FNCheckerboard::FNCheckerboard()':
fn_basic_generators.cpp:(.text+0x1f7): undefined reference to `FastNoise::Checkerboard* FastSIMD::New<FastNoise::Checkerboard>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_basic_generators.linuxbsd.template_release.x86_64.o): in function `FNSineWave::FNSineWave()':
fn_basic_generators.cpp:(.text+0x2b7): undefined reference to `FastNoise::SineWave* FastSIMD::New<FastNoise::SineWave>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_basic_generators.linuxbsd.template_release.x86_64.o): in function `FNPositionOutput::FNPositionOutput()':
fn_basic_generators.cpp:(.text+0x377): undefined reference to `FastNoise::PositionOutput* FastSIMD::New<FastNoise::PositionOutput>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_basic_generators.linuxbsd.template_release.x86_64.o): in function `FNDistanceToPoint::FNDistanceToPoint()':
fn_basic_generators.cpp:(.text+0x437): undefined reference to `FastNoise::DistanceToPoint* FastSIMD::New<FastNoise::DistanceToPoint>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_blends.linuxbsd.template_release.x86_64.o): in function `FNAdd::FNAdd()':
fn_blends.cpp:(.text+0xf0): undefined reference to `FastNoise::Add* FastSIMD::New<FastNoise::Add>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_blends.linuxbsd.template_release.x86_64.o): in function `FNSubtract::FNSubtract()':
fn_blends.cpp:(.text+0x250): undefined reference to `FastNoise::Subtract* FastSIMD::New<FastNoise::Subtract>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_blends.linuxbsd.template_release.x86_64.o): in function `FNMultiply::FNMultiply()':
fn_blends.cpp:(.text+0x3b0): undefined reference to `FastNoise::Multiply* FastSIMD::New<FastNoise::Multiply>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_blends.linuxbsd.template_release.x86_64.o): in function `FNDivide::FNDivide()':
fn_blends.cpp:(.text+0x510): undefined reference to `FastNoise::Divide* FastSIMD::New<FastNoise::Divide>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_cellular.linuxbsd.template_release.x86_64.o): in function `FNCellularValue::FNCellularValue()':
fn_cellular.cpp:(.text+0x6e6): undefined reference to `FastNoise::CellularValue* FastSIMD::New<FastNoise::CellularValue>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_cellular.linuxbsd.template_release.x86_64.o): in function `FNCellularDistance::FNCellularDistance()':
fn_cellular.cpp:(.text+0x976): undefined reference to `FastNoise::CellularDistance* FastSIMD::New<FastNoise::CellularDistance>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_cellular.linuxbsd.template_release.x86_64.o): in function `FNCellularLookup::FNCellularLookup()':
fn_cellular.cpp:(.text+0xf26): undefined reference to `FastNoise::CellularLookup* FastSIMD::New<FastNoise::CellularLookup>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_domain_warp.linuxbsd.template_release.x86_64.o): in function `FNDomainWarpGradient::FNDomainWarpGradient()':
fn_domain_warp.cpp:(.text+0x50): undefined reference to `FastNoise::DomainWarpGradient* FastSIMD::New<FastNoise::DomainWarpGradient>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_fractal.linuxbsd.template_release.x86_64.o): in function `FNFractal::new_fractal(int)':
fn_fractal.cpp:(.text+0x17d): undefined reference to `FastNoise::FractalFBm* FastSIMD::New<FastNoise::FractalFBm>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_fractal.linuxbsd.template_release.x86_64.o): in function `FNFractal::FNFractal(FNFractal::FractalType)':
fn_fractal.cpp:(.text+0x2f8): undefined reference to `FastNoise::FractalFBm* FastSIMD::New<FastNoise::FractalFBm>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_modifiers.linuxbsd.template_release.x86_64.o): in function `FNDomainScale::FNDomainScale()':
fn_modifiers.cpp:(.text+0x57): undefined reference to `FastNoise::DomainScale* FastSIMD::New<FastNoise::DomainScale>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/libmodule_gd_fastnoise2.linuxbsd.template_release.x86_64.a(fn_modifiers.linuxbsd.template_release.x86_64.o): in function `FNDomainOffset::FNDomainOffset()':
fn_modifiers.cpp:(.text+0x117): undefined reference to `FastNoise::DomainOffset* FastSIMD::New<FastNoise::DomainOffset>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Constant>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x25d1): undefined reference to `FastNoise::Constant* FastSIMD::New<FastNoise::Constant>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::White>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x266b): undefined reference to `FastNoise::White* FastSIMD::New<FastNoise::White>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Checkerboard>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2705): undefined reference to `FastNoise::Checkerboard* FastSIMD::New<FastNoise::Checkerboard>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::SineWave>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x279f): undefined reference to `FastNoise::SineWave* FastSIMD::New<FastNoise::SineWave>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::PositionOutput>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2839): undefined reference to `FastNoise::PositionOutput* FastSIMD::New<FastNoise::PositionOutput>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DistanceToPoint>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x28d3): undefined reference to `FastNoise::DistanceToPoint* FastSIMD::New<FastNoise::DistanceToPoint>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Value>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x296d): undefined reference to `FastNoise::Value* FastSIMD::New<FastNoise::Value>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Perlin>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2a07): undefined reference to `FastNoise::Perlin* FastSIMD::New<FastNoise::Perlin>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Simplex>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2aa1): undefined reference to `FastNoise::Simplex* FastSIMD::New<FastNoise::Simplex>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::OpenSimplex2>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2b3b): undefined reference to `FastNoise::OpenSimplex2* FastSIMD::New<FastNoise::OpenSimplex2>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::CellularValue>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2bd5): undefined reference to `FastNoise::CellularValue* FastSIMD::New<FastNoise::CellularValue>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::CellularDistance>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2c6f): undefined reference to `FastNoise::CellularDistance* FastSIMD::New<FastNoise::CellularDistance>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::CellularLookup>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2d09): undefined reference to `FastNoise::CellularLookup* FastSIMD::New<FastNoise::CellularLookup>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::FractalFBm>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2da3): undefined reference to `FastNoise::FractalFBm* FastSIMD::New<FastNoise::FractalFBm>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::FractalPingPong>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2e3d): undefined reference to `FastNoise::FractalPingPong* FastSIMD::New<FastNoise::FractalPingPong>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::FractalRidged>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2ed7): undefined reference to `FastNoise::FractalRidged* FastSIMD::New<FastNoise::FractalRidged>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainWarpGradient>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x2f71): undefined reference to `FastNoise::DomainWarpGradient* FastSIMD::New<FastNoise::DomainWarpGradient>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainWarpFractalProgressive>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x300b): undefined reference to `FastNoise::DomainWarpFractalProgressive* FastSIMD::New<FastNoise::DomainWarpFractalProgressive>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainWarpFractalIndependant>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x30a5): undefined reference to `FastNoise::DomainWarpFractalIndependant* FastSIMD::New<FastNoise::DomainWarpFractalIndependant>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainScale>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x313f): undefined reference to `FastNoise::DomainScale* FastSIMD::New<FastNoise::DomainScale>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainOffset>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x31d9): undefined reference to `FastNoise::DomainOffset* FastSIMD::New<FastNoise::DomainOffset>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainRotate>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3273): undefined reference to `FastNoise::DomainRotate* FastSIMD::New<FastNoise::DomainRotate>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::SeedOffset>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x330d): undefined reference to `FastNoise::SeedOffset* FastSIMD::New<FastNoise::SeedOffset>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Remap>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x33a7): undefined reference to `FastNoise::Remap* FastSIMD::New<FastNoise::Remap>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::ConvertRGBA8>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3441): undefined reference to `FastNoise::ConvertRGBA8* FastSIMD::New<FastNoise::ConvertRGBA8>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Add>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x34db): undefined reference to `FastNoise::Add* FastSIMD::New<FastNoise::Add>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Subtract>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3575): undefined reference to `FastNoise::Subtract* FastSIMD::New<FastNoise::Subtract>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Multiply>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x360f): undefined reference to `FastNoise::Multiply* FastSIMD::New<FastNoise::Multiply>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Divide>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x36a9): undefined reference to `FastNoise::Divide* FastSIMD::New<FastNoise::Divide>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Min>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3743): undefined reference to `FastNoise::Min* FastSIMD::New<FastNoise::Min>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Max>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x37dd): undefined reference to `FastNoise::Max* FastSIMD::New<FastNoise::Max>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::MinSmooth>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3877): undefined reference to `FastNoise::MinSmooth* FastSIMD::New<FastNoise::MinSmooth>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::MaxSmooth>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3911): undefined reference to `FastNoise::MaxSmooth* FastSIMD::New<FastNoise::MaxSmooth>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Fade>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x39ab): undefined reference to `FastNoise::Fade* FastSIMD::New<FastNoise::Fade>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::Terrace>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3a45): undefined reference to `FastNoise::Terrace* FastSIMD::New<FastNoise::Terrace>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::PowFloat>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3adf): undefined reference to `FastNoise::PowFloat* FastSIMD::New<FastNoise::PowFloat>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::PowInt>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3b79): undefined reference to `FastNoise::PowInt* FastSIMD::New<FastNoise::PowInt>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::DomainAxisScale>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3c13): undefined reference to `FastNoise::DomainAxisScale* FastSIMD::New<FastNoise::DomainAxisScale>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::AddDimension>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3cad): undefined reference to `FastNoise::AddDimension* FastSIMD::New<FastNoise::AddDimension>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::RemoveDimension>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3d47): undefined reference to `FastNoise::RemoveDimension* FastSIMD::New<FastNoise::RemoveDimension>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::GeneratorCache>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3de1): undefined reference to `FastNoise::GeneratorCache* FastSIMD::New<FastNoise::GeneratorCache>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
/usr/bin/ld: modules/gd_fastnoise2/FastNoise2/src/libFastNoise.a(Metadata.o): in function `FastNoise::MetadataT<FastNoise::OpenSimplex2S>::CreateNode(FastSIMD::eLevel) const':
Metadata.cpp:(.text+0x3e7b): undefined reference to `FastNoise::OpenSimplex2S* FastSIMD::New<FastNoise::OpenSimplex2S>(FastSIMD::eLevel, void* (*)(unsigned long, unsigned long))'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.linuxbsd.template_release.x86_64.mono] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:12.112]
```

**Comment by zak-grumbles** [comment]
Thanks for reporting this, I'll take a look as soon as I get a chance.

**Comment by zak-grumbles** [comment]
@The64thGamer Fix is in for this via 41d757e3ebdef52ee6b5e2f27c660f6164dfe39f . I don't currently have access to a linux system so please let me know if you have further issues.

**Comment by The64thGamer** [comment]
This commit does correct it finding the file, but similar to me just renaming the file to match- compiling Godot still brings that second error where it doesn't find any of the functions and refuses to finish compiling.

**Comment by zak-grumbles** [comment]
I believe I found the root cause of the issue. Over in my [FastNoise2 fork](https://github.com/zak-grumbles/FastNoise2), the Sconscript file wasn't properly accounting for compilers that were neither msvc or llvm. It should be fixed by [this commit](https://github.com/zak-grumbles/FastNoise2) in that repo, and by 9b80636a5bd64eb23130ae31021dc70e2e946649 in this one. 

I tested the fix by spinning up an ubuntu vm and compiling godot successfully. I'll leave the issue open for now, but let me know if that resolves it for you, @The64thGamer

**Comment by The64thGamer** [comment]
This does correctly compile the binary now. However when compiling for C# and generating the C# glue in the second step after compilation, another error pops up.
```
ERROR: Return type is reference but hint is not 'PROPERTY_HINT_RESOURCE_TYPE'. Are you returning a reference type by pointer? Method: 'FNFractal.new_fractal'.
   at: _populate_object_type_interfaces (modules/mono/editor/bindings_generator.cpp:3198)
ERROR: Failed to generate object type interfaces
   at: _initialize (modules/mono/editor/bindings_generator.cpp:4174)
ERROR: Failed to initialize the bindings generator
```
Though it could be unrelated to this if you would like me to post this as a separate issue in the tracker.

**Comment by zak-grumbles** [comment]
Apologies for the delay. I'm fine with tracking it here. I'm not as familiar with C# in Godot so it may take me a bit longer to diagnose this one but I'll look into it. 👍

**Comment by zak-grumbles** [comment]
Alright, I dug out an old linux laptop I had and got this resolved. I'm going to close this ticket but if you run into anything else feel free to open a new one.

I'll be doing some more testing later this week to make sure these changes didn't result in further problems.

**Comment by zak-grumbles** [comment]
Fixed by 8893e40e2d8444af514dae0c33835360400b5d2b

**Comment by The64thGamer** [comment]
No issues on my end! Thanks a ton for adding this.
